components/create_post/create_post.jsx

```diff
@@ -596,7 +616,7 @@ export default class CreatePost extends React.Component {
             }
         }
 
-        this.setState({serverError: message});
+        this.setState({serverError: {message}});
```

Comment:
Hey @mickmister Can you help me out on this? Why are we setting the `message` on a key in serverError object here?

Reply:
Yes, this code is incorrect. Before this PR's introduction, the `serverError` state variable was of type `string`, and was used to represent the error as a message. With this invalid slash command issue fix, we need to store the entire error in order to access `serverError.server_error_code`. To follow the convention of the js error's message being stored at `error.message`, I made an object version of the error here where the message can be accessed the same way. In reality, this should be a real error.

Looking at the rest of the code in this method, it looks like we are extracting out the `message` string when we have a real server error, and discarding the actual error object given. I propose that instead of:

```
let message = err;
if (message && typeof message !== 'string') {    
    message = err.message;
}

this.setState({serverError: {message}});
```

We do:

```
let serverError = err;
if (typeof err === 'string') {
    serverError = new Error(err);
}

this.setState({serverError});
```

That way, we have a real error to work with no matter what, and we preserve all context provided by the original server error. How does this sound?

Reply:
That makes a lot of sense. Sounds good to me 👍.

Reply:
@sudheerDev all requested changes resolved